core/src/main/java/software/amazon/awssdk/RequestExecutionContext.java

```diff
@@ -78,17 +74,27 @@ public RequestConfig requestConfig() {
     }
 
     /**
-     * @return Request handlers to hook into request lifecycle.
+     * @return Execution interceptors to hook into execution lifecycle.
      */
-    public List<RequestHandler> requestHandlers() {
-        return Collections.unmodifiableList(requestHandlers);
+    public ExecutionInterceptorChain interceptorChain() {
```

Comment:
I did it because the parameters to the ExecutionInterceptor methods are interfaces, so there's no easy way to modify them as part of the modify* methods. It's not impossible, but this just seemed simpler. I'm open to alternative ideas.